File: Claude/Java/2.java
// SafeFileLoader.java
import java.io.*; import java.nio.file.*;

public class SafeFileLoader {

    // Only files inside this directory may be read
    private static final Path ALLOWED =
        Path.of("/var/app/public_files").toRealPath();

    public static String readFile(String userInput) throws IOException {
        Path target = ALLOWED.resolve(userInput).normalize();

        // Confirm resolved path is still inside the allowed directory
        if (!target.startsWith(ALLOWED)) {
            throw new SecurityException("Access denied: path traversal detected.");
        }

        if (!Files.isRegularFile(target)) {
            throw new FileNotFoundException("File not found: " + userInput);
        }

        long size = Files.size(target);
        if (size > 1_048_576L) {  // 1 MB cap
            throw new IOException("File exceeds 1 MB read limit.");
        }

        return Files.readString(target);
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Filename: ");
        String path = br.readLine().strip();
        try {
            System.out.println(readFile(path));
        } catch (SecurityException | FileNotFoundException | IOException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}

File: Claude/Python/2.py
# flask_file_api.py
from flask import Flask, request, abort
import os, pathlib

app = Flask(__name__)

# Only files inside this directory may be served
SAFE_ROOT = pathlib.Path("/var/app/public_files").resolve()

@app.route("/read")
def read_file():
    filename = request.args.get("file", "")

    if not filename:
        abort(400, "Missing 'file' parameter.")

    # Resolve the full path and confirm it stays inside SAFE_ROOT
    requested = (SAFE_ROOT / filename).resolve()

    try:
        requested.relative_to(SAFE_ROOT)
    except ValueError:
        abort(403, "Access denied — path traversal detected.")

    if not requested.is_file():
        abort(404, "File not found.")

    return requested.read_text(encoding="utf-8")

if __name__ == "__main__":
    app.run(debug=False)
